libfastpm/thermalvelocity.c

```diff
@@ -284,16 +294,24 @@ fastpm_ncdm_init_create(
     nid->n_shells = n_shells;
     nid->n_side = n_side;
     nid->lvk = lvk;
-
+    nid->ncdm_sphere_scheme = ncdm_sphere_scheme;
+    
     /* this is the total number of velocity vectors produced */
-    nid->n_split = 12 * n_shells * n_side * n_side;
+    if (ncdm_sphere_scheme == FASTPM_NCDM_SPHERE_HEALPIX){
```

Comment:
Perhaps use a switch case -- compiler will complain for unhandled cases.

Reply:
Makes sense! I'll fix when Cori is back